### ISSUE TITLE
carousel: no overflow leads to large right margin

### ISSUE THREAD

**Issue by jaketrent** [pluralsight/design-system#625]
From PD, Jordan Cooper
All the time (not just scroll)
Doesn't happen in 1.1.4.
Happens later.
Maybe try a case with a ton of items.
Or a single item with super wide width

**Comment by jaketrent** [comment]
A number of things I've found and fixed:

- Site rendering of Carousel after ActionMenu click was because top-level state was changing and needed to re-render the whole Carousel.  This was solved by storing ActionMenu toggle state per element.  Now vdom updates happen lower (inside the single array element).  As a side-effect of storing state lower, the upper component now doesn't control menu toggling.  Thus, the feature in the example of closing all ActionMenus when paging the Carousel needed to be re-implemented.  Looking at the ActionMenu, turns out we had an Overlay, but it wasn't used anywhere internally (but also exported).  I change ActionMenu to render Overlay when an onClose prop is provided.  The prop is a good conditional, because otherwise a "close the action menu" trigger from the ActionMenu can't notify any external entity controlling the actual rendering (because it's not controlled internally).
- Scrolling to the top of the page on ActionMenu toggling was happening because of a combo of 2 things: Position paints the shown element on the screen, then has refs, then measures it, then positions it.  ActionMenu focuses itself when rendered.  Thus rendering initially at 0,0, the focus would scroll to the top.  Now, focus on ActionMenu happens on nextTick.  It doesn't impact UX, afaict, and it supports this Position combo use case.
